### PR TITLE
allow derivingVia using quantified constraints to be expressible!

### PR DESCRIPTION
**Pull request by cartazio**

### CODE REVIEW

**Comment by cartazio** [comment]
https://gitlab.haskell.org/ghc/ghc/issues/17767#note_251123 for how this came about 

and 
```haskell
newtype MyDouble = MYD Double

type MGetCoerT m = (forall p q. (MonadGet m, Coercible p q) => Coercible (m p) (m q) :: Constraint)
deriving via Double instance MGetCoerT m => Serial MyDouble

```

**Comment by cartazio** [comment]
so that folks can see how to write stuff

**Comment by cartazio** [comment]
@RyanGlScott @ekmett would this be a minor or Major bump? and also a friend of ours would like it on hackage so he can use it ;)

**Comment by ekmett** [comment]
I'd probably major bump it to be safe.

**Comment by cartazio** [comment]
yeah, that extra constraint might impact someone somehwere

**Comment by cartazio** [comment]
@RyanGlScott  :)

File: src/Data/Bytes/Get.hs [hunk 1]
```diff
@@ -3,6 +3,10 @@
 {-# LANGUAGE FlexibleContexts #-}
 {-# LANGUAGE DefaultSignatures #-}
 {-# LANGUAGE Trustworthy #-}
+#if MIN_VERSION_base(4,12,0)
```
Comment (phadej) [inline src/Data/Bytes/Get.hs]: this should be on `__GLASGOW_HASKELL__`, quantified constraints isn't `base` feature.

**Comment by cartazio** [comment]
true, i suppose that would better, the bound would be 8 6?

**Comment by cartazio** [comment]
the magical future where ghc and base versions aren't coupled :)

File: src/Data/Bytes/Get.hs (hunk 1 above)
Comment (RyanGlScott) [reply src/Data/Bytes/Get.hs]: I've addressed this in commit 79554390ef3b7445460a830d65caa0d54f26a60a.

**Comment by RyanGlScott** [comment]
I'll plan to make a new Hackage release of `bytes` sometime next week alongside other @ekmett libraries that need major version bumps (namely, `ad`, `constraints`, `lens`, and `linear`).

**Comment by RyanGlScott** [comment]
I've released [`bytes-0.17`](http://hackage.haskell.org/package/bytes-0.17) to Hackage with these changes.

**Comment by cartazio** [comment]
Yay and thank you!

**Pull request merged**